californium-core/src/main/java/org/eclipse/californium/core/network/MessageExchangeStore.java

```diff
@@ -181,6 +184,15 @@
 	boolean isEmpty();
 
 	/**
+	 * Checks if there are any exchanges currently being registered in this
+	 * store and dump these exchanges. Intended to be used for more details in
+	 * log when testing.
+	 * 
+	 * @return {@code true} if there no exchanges registered.
+	 */
+	boolean isEmptyWithDump();
```

Comment:
I suppose `MessageExchangeStore` interface was introduced to let users implement their own `MessageExchangeStore`. This adds flexibility, so it  sounds good idea.

An interface is a contract with our users, if we add this method we force users to implement it although we don't use it. This seems a bad idea.  Especially because isEmptyWithDump has a really limited scope of usefulness.

FMPOV, even having `isEmptyWithDump` in `InMemoryExchangeStore` sounds strange. I will just go for adding getter for  `ExchangeMaps (only in implementation)` and let `Tests` classes the responsibility to log it.

> The arguments for placing the isEmptyWithDump are the same as for the already available isEmpty

I don't know the arguments about why we add `isEmpty` in the interface. It's seems that we used it only in tests. I maybe validated a PR with this addition. Maybe I didn't notice it because having an `isEmpty` method on a `Store` interface is usual.

I share my thought with you. If you still disagree, too bad :). I suppose my arguments was not strong enough.

Anyway, I don't think that thinking about "what is part of the API" or "what is the responsability of a class" is a "waste of time". No offense but what kind of feedbacks would you expect when you open a PR ? Personally I check for bugs, code readability and code organization :)